diff -Nur ffcall-20090629.orig/avcall/Makefile.in ffcall-20090629/avcall/Makefile.in
--- ffcall-20090629.orig/avcall/Makefile.in	2009-04-27 18:44:13.000000000 +0200
+++ ffcall-20090629/avcall/Makefile.in	2009-06-29 22:19:04.000000000 +0200
@@ -72,6 +72,12 @@
 avcall-sparc.s : $(srcdir)/avcall-sparc.S
 	$(CPP) $(ASPFLAGS) $(srcdir)/avcall-sparc.S | grep -v '^ *#line' | grep -v '^#' | sed -e 's,% ,%,g' -e 's,//.*$$,,' -e 's,\$$,#,g' > avcall-sparc.s
 
+avcall-sparcv9.lo : avcall-sparcv9.s
+	$(LIBTOOL_COMPILE) $(CC) @GCC_X_NONE@ -c avcall-sparcv9.s
+
+avcall-sparcv9.s : $(srcdir)/avcall-sparc.S
+	$(CPP) $(ASPFLAGS) $(srcdir)/avcall-sparc.S | grep -v '^ *#line' | grep -v '^#' | sed -e 's,% ,%,g' -e 's,//.*$$,,' -e 's,\$$,#,g' > avcall-sparcv9.s
+
 avcall-sparc64.lo : avcall-sparc64.s
 	$(LIBTOOL_COMPILE) $(CC) @GCC_X_NONE@ -c avcall-sparc64.s
 
diff -Nur ffcall-20090629.orig/callback/trampoline_r/Makefile.in ffcall-20090629/callback/trampoline_r/Makefile.in
--- ffcall-20090629.orig/callback/trampoline_r/Makefile.in	2009-04-27 18:44:13.000000000 +0200
+++ ffcall-20090629/callback/trampoline_r/Makefile.in	2009-06-29 22:21:56.000000000 +0200
@@ -91,6 +91,12 @@
 cache-sparc.s : $(srcdir)/cache-sparc.S
 	$(CPP) $(ASPFLAGS) $(srcdir)/cache-sparc.S | grep -v '^ *#line' | grep -v '^#' | sed -e 's,% ,%,g' -e 's,//.*$$,,' -e 's,\$$,#,g' > cache-sparc.s
 
+cache-sparcv9.lo : cache-sparcv9.s
+	$(LIBTOOL_COMPILE) $(CC) @GCC_X_NONE@ -c cache-sparcv9.s
+
+cache-sparcv9.s : $(srcdir)/cache-sparcv9.S
+	$(CPP) $(ASPFLAGS) $(srcdir)/cache-sparc.S | grep -v '^ *#line' | grep -v '^#' | sed -e 's,% ,%,g' -e 's,//.*$$,,' -e 's,\$$,#,g' > cache-sparcv9.s
+
 cache-sparc64.lo : cache-sparc64.s
 	$(LIBTOOL_COMPILE) $(CC) @GCC_X_NONE@ -c cache-sparc64.s
 
diff -Nur ffcall-20090629.orig/callback/vacall_r/Makefile.in ffcall-20090629/callback/vacall_r/Makefile.in
--- ffcall-20090629.orig/callback/vacall_r/Makefile.in	2009-04-27 18:44:14.000000000 +0200
+++ ffcall-20090629/callback/vacall_r/Makefile.in	2009-06-29 22:20:41.000000000 +0200
@@ -72,6 +72,12 @@
 vacall-sparc.s : $(srcdir)/vacall-sparc.S
 	$(CPP) $(ASPFLAGS) $(srcdir)/vacall-sparc.S | grep -v '^ *#line' | grep -v '^#' | sed -e 's,% ,%,g' -e 's,//.*$$,,' -e 's,\$$,#,g' > vacall-sparc.s
 
+vacall-sparcv9.lo : vacall-sparcv9.s
+	$(LIBTOOL_COMPILE) $(CC) @GCC_X_NONE@ -c vacall-sparcv9.s
+
+vacall-sparcv9.s : $(srcdir)/vacall-sparc.S
+	$(CPP) $(ASPFLAGS) $(srcdir)/vacall-sparc.S | grep -v '^ *#line' | grep -v '^#' | sed -e 's,% ,%,g' -e 's,//.*$$,,' -e 's,\$$,#,g' > vacall-sparcv9.s
+
 vacall-sparc64.lo : vacall-sparc64.s
 	$(LIBTOOL_COMPILE) $(CC) @GCC_X_NONE@ -c vacall-sparc64.s
 
diff -Nur ffcall-20090629.orig/trampoline/Makefile.in ffcall-20090629/trampoline/Makefile.in
--- ffcall-20090629.orig/trampoline/Makefile.in	2009-04-27 18:44:14.000000000 +0200
+++ ffcall-20090629/trampoline/Makefile.in	2009-06-29 22:24:25.000000000 +0200
@@ -83,6 +83,12 @@
 cache-sparc.s : $(srcdir)/cache-sparc.S
 	$(CPP) $(ASPFLAGS) $(srcdir)/cache-sparc.S | grep -v '^ *#line' | grep -v '^#' | sed -e 's,% ,%,g' -e 's,//.*$$,,' -e 's,\$$,#,g' > cache-sparc.s
 
+cache-sparcv9.o : cache-sparcv9.s
+	$(CC) @GCC_X_NONE@ -c cache-sparcv9.s
+
+cache-sparcv9.s : $(srcdir)/cache-sparc.S
+	$(CPP) $(ASPFLAGS) $(srcdir)/cache-sparc.S | grep -v '^ *#line' | grep -v '^#' | sed -e 's,% ,%,g' -e 's,//.*$$,,' -e 's,\$$,#,g' > cache-sparcv9.s
+
 cache-sparc64.o : cache-sparc64.s
 	$(CC) @GCC_X_NONE@ -c cache-sparc64.s
 
diff -Nur ffcall-20090629.orig/vacall/Makefile.in ffcall-20090629/vacall/Makefile.in
--- ffcall-20090629.orig/vacall/Makefile.in	2009-04-27 18:44:14.000000000 +0200
+++ ffcall-20090629/vacall/Makefile.in	2009-06-29 22:26:28.000000000 +0200
@@ -63,6 +63,12 @@
 vacall-sparc.s : $(srcdir)/vacall-sparc.S
 	$(CPP) $(ASPFLAGS) $(srcdir)/vacall-sparc.S | grep -v '^ *#line' | grep -v '^#' | sed -e 's,% ,%,g' -e 's,//.*$$,,' -e 's,\$$,#,g' > vacall-sparc.s
 
+vacall-sparcv9.o : vacall-sparcv9.s
+	$(CC) @GCC_X_NONE@ -c vacall-sparcv9.s
+
+vacall-sparcv9.s : $(srcdir)/vacall-sparc.S
+	$(CPP) $(ASPFLAGS) $(srcdir)/vacall-sparc.S | grep -v '^ *#line' | grep -v '^#' | sed -e 's,% ,%,g' -e 's,//.*$$,,' -e 's,\$$,#,g' > vacall-sparcv9.s
+
 vacall-sparc64.o : vacall-sparc64.s
 	$(CC) @GCC_X_NONE@ -c vacall-sparc64.s
 
