--- libffcall-2.0/callback/Makefile.in.orig	2017-09-10 13:48:38.000000000 +0200
+++ libffcall-2.0/callback/Makefile.in	2017-12-04 19:00:05.515956833 +0100
@@ -76,6 +76,10 @@
 callback-compat.lo : $(srcdir)/callback-compat.c ../config.h
 	$(LIBTOOL_COMPILE) $(CC) $(INCLUDES) $(CPPFLAGS) $(CFLAGS) -c $(srcdir)/callback-compat.c
 
+vacall_r/libvacall.la: all-subdirs
+
+trampoline_r/libtrampoline.la: all-subdirs
+
 libcallback.la : vacall_r/libvacall.la trampoline_r/libtrampoline.la callback-libapi.lo callback-compat.lo
 	$(LIBTOOL_LINK) $(CC) -o libcallback.la -rpath $(libdir) -no-undefined -export-symbols-regex $(LIBCALLBACK_EXPORTED_SYMBOLS_REGEX) -version-info $(LIBCALLBACK_VERSION_INFO) vacall_r/vacall.lo vacall_r/vacall-libapi.lo vacall_r/vacall-structcpy.lo trampoline_r/*.lo callback-libapi.lo callback-compat.lo ../gnulib-lib/libgnu.la $(LDFLAGS) $(LTLIBTHREAD)
 
--- libffcall-2.0/Makefile.in.orig	2017-09-10 13:49:17.000000000 +0200
+++ libffcall-2.0/Makefile.in	2017-12-04 20:16:10.922571372 +0100
@@ -65,6 +65,10 @@
 ffcall-version.lo : $(srcdir)/ffcall-version.c ffcall-version.h config.h
 	$(LIBTOOL_COMPILE) $(CC) $(INCLUDES) $(CPPFLAGS) $(CFLAGS) -c $(srcdir)/ffcall-version.c
 
+avcall/avcall.lo avcall/avcall-libapi.lo avcall/avcall-structcpy.lo: all-subdirs
+
+callback/vacall_r/libvacall.la callback/trampoline_r/libtrampoline.la callback/callback-libapi.lo: all-subdirs
+
 libffcall.la : ffcall-version.lo avcall/avcall.lo avcall/avcall-libapi.lo avcall/avcall-structcpy.lo callback/vacall_r/libvacall.la callback/trampoline_r/libtrampoline.la callback/callback-libapi.lo
 	$(LIBTOOL_LINK) $(CC) -o libffcall.la -rpath $(libdir) -no-undefined -export-symbols-regex $(LIBFFCALL_EXPORTED_SYMBOLS_REGEX) -version-info $(LIBFFCALL_VERSION_INFO) ffcall-version.lo avcall/avcall.lo avcall/avcall-libapi.lo avcall/avcall-structcpy.lo callback/vacall_r/vacall.lo callback/vacall_r/vacall-libapi.lo callback/vacall_r/vacall-structcpy.lo callback/trampoline_r/*.lo callback/callback-libapi.lo gnulib-lib/libgnu.la $(LDFLAGS) $(LTLIBTHREAD)
 
